src/services/armService.ts

```diff
@@ -0,0 +1,62 @@
+import Serverless from "serverless";
+import { Deployment } from "@azure/arm-resources/esm/models";
+import { BaseService } from "./baseService";
+import { ResourceManagementClient } from "@azure/arm-resources";
+import { Guard } from "../shared/guard";
+
+export interface ArmTemplateGenerator {
+  generate(): any;
+}
+
+export class ArmService extends BaseService {
+  private resourceClient: ResourceManagementClient;
+
+  public constructor(serverless: Serverless) {
+    super(serverless);
+    this.resourceClient = new ResourceManagementClient(this.credentials, this.subscriptionId);
+  }
+
+  public async createTemplate(type: string): Promise<any> {
```

Comment:
Example of dynamically merging in the templates and optionally exposing and merging APIM component.